experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using Microsoft.Bot.Builder;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using System.Linq;
+using LivePersonAgentBot.Bots;
+using System.Net.Http;
+using Newtonsoft.Json;
+using System.Text;
+using System.Collections.Concurrent;
+using Newtonsoft.Json.Linq;
+
+namespace LivePersonAgentBot
+{
+    public class HandoffMiddleware : Microsoft.Bot.Builder.IMiddleware
+    {
+        public HandoffMiddleware()
+        {
+        }
+
+        public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default)
+        {
+            turnContext.OnSendActivities(async (sendTurnContext, activities, nextSend) =>
+            {
+                Activity handOffEventActivity = activities.Where((activity) => activity.Type == ActivityTypes.Event && activity.Name == HandoffEventNames.InitiateHandoff).FirstOrDefault();
```

Comment:
This statement is quite complex, involving LINQ, delegates, operator precedence, and default values. 

Consider simplifying for clarity. 

Also, there's no guarantee that activities won't be null, as it wasn't checked - so this may all throw a NRE.